analysis/analysisfunctions.py

```diff
@@ -209,35 +209,47 @@ def get_experiment_table(faultgroup, faultname, tablename):
     table = node._f_get_child(tablename)
     return pd.DataFrame(table.read())
 
-def get_experiment_table_expanded(filehandle, faultname, tablename, keyword):
+def get_experiment_table_expanded(filehandle, faultname, tablename, keywords):
     """
     Get the experiment table and recombine it with the goldenrun data
     TODO: This data must be checked if not to much is included
     """
     golden_table = get_experiment_table(filehandle.root, "Goldenrun", tablename)
     exp_table = get_experiment_table(filehandle.root.fault, faultname, tablename)
-    tmp = exp_table[keyword]
-    idx = None
-    for t in tmp:
-        idt = golden_table.index[golden_table[keyword] == t]
-        if idx is None:
-            idx = idt
+    idxs = None
+    for keyword in keywords:
+        tmp = exp_table[keyword]
+        idx = None
+        for t in tmp:
+            idt = golden_table.index[golden_table[keyword] == t]
+            if idx is None:
+                idx = idt
+            else:
+                idx.append(idt)
+        if idxs is None:
+            idxs = idx
         else:
-            idx.append(idt)
-    if idx is not None:
-        golden_table.drop(idx, inplace=True)
+            idxs = idxs.intersection(idx)
+    if idxs is not None:
+        golden_table.drop(idxs, inplace=True)
     data = [exp_table, golden_table]
     return pd.concat(data).to_dict("records")
 
 def get_experiment_tbinfo(faultgroup, faultname):
     return get_experiment_table(faultgroup, faultname, "tbinfo")
 
 def get_experiment_tbinfo_expanded(filehandle, faultname):
-    return get_experiment_table_expanded(filehandle, faultname, "tbinfo", "identity")
+    return get_experiment_table_expanded(filehandle, faultname, "tbinfo", ["identity"])
```

Comment:
No, this is the commit that introduced the change of the generic function. Before this commit the generic function took in a keyword to operate on. Because Meminfo needs two changed generic function to handle mutliple keywords. Changed old functions to be compliant. If i would change them in the original commit, i would break the functionality.

Reply:
Maybe I misunderstand something, but is the function `get_experiment_table_expanded` not added within this PR?

It seems to be added [here](https://github.com/Fraunhofer-AISEC/archie/blob/9b6541c683050a31f43510e8839c517c966fb1a4/analysis/analysisfunctions.py#L212) and extended [here](https://github.com/Fraunhofer-AISEC/archie/blob/e93ddb4ac3c53ef63627978a4aaea4aef33a4507/analysis/analysisfunctions.py#L212).

Why not add it at once with support for `keywords`?

Reply:
That was not the point. These commits were done during developement. I did not think of using multiple keywords until i implemented the memory function. But i have 2 problems with the changing of a commit. Firstly i have no idea how to alter a commit as i never felt the need to do so. Do you user git commit --amend or is it something else? Secondly i am not sure whats the benefit of overwriting a commit if it extends a in a previous commit added function. For the first and second commit i am certain that both are working, but if i overwrite the first commit with content introduced in the second it might not if i forget something. What is the benefit of doing this? When not overwriting the commit it is possible to see sort of a history of the development process.

Reply:
> But i have 2 problems with the changing of a commit. Firstly i have no idea how to alter a commit as i never felt the need to do so. Do you user git commit --amend or is it something else? 

Best is to use `git rebase --interactive HEAD~<# of commits to go back>` and for example the `fixup` option. `fixup` combines commits into one. So no manually effort needed.

> That was not the point. These commits were done during developement. I did not think of using multiple keywords until i implemented the memory function. 

> Secondly i am not sure whats the benefit of overwriting a commit if it extends a in a previous commit added function. [...] What is the benefit of doing this? When not overwriting the commit it is possible to see sort of a history of the development process.

IMO the commit history of a PR must not resemble the development process, but the outcome so to say. So if something seems not to be a good choice during development or if typos or style issues are found (and the PR was not yet merged into master), the best is to change prior commits in the feature branch.